src/dropdown.ts

```diff
@@ -2444,17 +2681,39 @@ export default class Dropdown extends LitElement implements FormControl {
   }
 
   #show() {
-    // `#show` is called whenever `open` is set. And `open` can be set arbitrarily
-    // by the consumer. Rather than guarding against calling `#show` everywhere,
-    // Floating UI is simply cleaned up every time `#show` is called.
+    // `#show()` is called whenever `open` is set. And `open` can be set arbitrarily
+    // by the consumer. Rather than guarding against calling `#show()` everywhere,
+    // Floating UI is simply cleaned up every time `#show()` is called.
     this.#cleanUpFloatingUi?.();
 
-    // This is done now instead of after Floating UI does its thing because the
-    // user may begin arrowing immediately to another option option or may have
-    // arrowed to open Dropdown and then accidentally arrowed again.
-    if (this.#previouslyActiveOption) {
+    if (this.#optionElementsNotHidden) {
+      this.itemCount = this.#optionElementsNotHidden.length;
```

Comment:
`this.itemCount` wasn't being set before on `#show()`. It didn't result in a bug because the count isn't presented to screenreaders until the user begins filtering. But the count should always be correct nonetheless. 

I [added](https://github.com/CrowdStrike/glide-core/pull/929/files#diff-749aecf3d8907f729dad99b541e170b750aeed9ebcfabe5a57b850521515213cR1596-R1607) a test for this.